modules/core/src/smithy4s/http/HttpEndpoint.scala

```diff
@@ -23,8 +23,14 @@ import smithy4s.http.internals.SchemaVisitorPathEncoder
 
 trait HttpEndpoint[I] {
   // Returns a list of path segments that should be appended to the base URL. These are not URL-encoded.
+  @deprecated(
+    "Path segments should be encoded in a specific way per the Smithy spec, so use encodedPath instead",
+    "v0.18.34"
```

Comment:
it's gonna be .40 now, and maybe without the `v`

Reply:
also, should we actually deprecate it, if we keep using `path` in the case of `if rawHttpLabelValues`? I suppose the plan is to remove `rawHttpLabelValues` later and make `true` the default?